tests/virt_autotest/virsh_snapshot.pm

```diff
@@ -0,0 +1,54 @@
+# Copyright (C) 2019 SUSE LLC
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, see <http://www.gnu.org/licenses/>.
+
+# Summary: Test VM snapshot using virsh (create - restore - delete)
+# Maintainer: Leon Guo <xguo@suse.com>
+ 
+use strict;
+use warnings;
+use testapi;
+use base "virt_autotest_base";
+
+sub run {
+
+    my $guest = script_output("virsh list --all| sed -n 3p | awk {'print \$2'}");
+
```

Comment:
As yesterday mentioned, please do it the way similar to how qam tests. Loop all the guests for the test operations, just like https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/tests/virtualization/xen/hotplugging.pm. Both QAM and us can save efforts when integrating each other if we follow similar style.